core/src/main/java/org/infinispan/container/DataContainer.java

```diff
@@ -90,10 +90,19 @@
    InternalCacheEntry<K, V> remove(Object k);
 
    /**
-    *
+    * Returns how many entries are present in the data container without counting entries that are currently expired
     * @return count of the number of entries in the container excluding expired entries
+    * @implSpec
+    * Default method invokes the {@link #iterator()} method and just counts entries.
     */
-   int size();
+   default int size() {
+      int size = 0;
+      // We have to loop through to make sure to remove expired entries
+      for (InternalCacheEntry<K, V> ignore : this) {
```

Comment:
The problem I found with that is I have to use an `AtomicInteger` instead of int and it can't early terminate due to having more than `Integer.MAX_VALUE` entries.